### PR TITLE
Use tasty-golden for golden tests and ditch vendored hspec-expectations package.

### LINKED ISSUE #120: Use tasty + tasty-golden for parse-examples tests.
These diffs are a prime example of golden testing: we want to match diff output exactly. If we use `tasty` as our prime test runner, we can use `tasty-golden` to set up a more accurate test of meaning (rather than just using `shouldBe` on a big string), and we’ll get pretty diff output without having to keep `hspec-expectations-pretty-diff` around (#116).

### PR DESCRIPTION
**Pull request by patrickt**
This patch:
* divests from [our fork](https://github.com/rewinfrey/hspec-expectations-pretty-diff) of `hspec-expectations-pretty-diff`, which partially unblocks #16.
* pulls in https://github.com/feuerbach/tasty as a test harness/runner
* pulls in https://github.com/feuerbach/tasty-golden to provide golden tests
* pulls in https://github.com/mitchellwrosen/tasty-hspec for backcompat with our hspec suite (after an abortive attempt to convert the hspec suite, which produced a diff of truly depressing size)
* uses `unsafePerformIO` in places where it is somewhat unavoidable (I defend `unsafePerformIO`’s use in tests)
* gets good diff output because `tasty-golden` lets us provide our own diff command, and attempts to use [`jd`](https://github.com/josephburnett/jd) for JSON comparisons.
* makes all the other unit tests use normal `hspec-expectations`
* adds some missing version bounds
* removes a ton of unused imports in the specs

Fixes #120.

### CODE REVIEW

File: test/Integration/Spec.hs [hunk 1]
```diff
@@ -1,36 +1,53 @@
+{-# LANGUAGE ImplicitParams, LambdaCase, NamedFieldPuns #-}
```
[the remark this answers is not in the document]
Comment (patrickt) [reply test/Integration/Spec.hs]: (I do think implicit params are a better solution than plumbing parameters everywhere or `Reader` + umpteen calls to `pure`. But we should arguably do the `let ?session` in the top-level Spec.hs file.)

File: semantic.cabal [hunk 2]
```diff
@@ -274,7 +274,7 @@ library
   autogen-modules:     Paths_semantic
   other-modules:       Paths_semantic
   build-depends:       base >= 4.12 && < 5
-                     , ansi-terminal ^>= 0.8.2
+                     , ansi-terminal >= 0.8.2 && <1
```
Comment (robrix) [inline semantic.cabal]: Pourquoi?

File: semantic.cabal [hunk 3]
```diff
@@ -340,6 +340,7 @@ test-suite test
   import:              haskell, dependencies, executable-flags
   type:                exitcode-stdio-1.0
   hs-source-dirs:      test
+  ghc-options:         -Wunused-imports
```
Comment (robrix) [inline semantic.cabal]: Can we apply this in a common stanza?

File: test/Integration/Spec.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (robrix) [reply test/Integration/Spec.hs]: _u wot m8_

The purist in me would prefer a `Reader` effect, but I see the appeal. Tell you what: _you_ maintain it and we’re good 😉

File: test/Spec.hs [hunk 4]
```diff
@@ -37,43 +37,70 @@ import qualified Semantic.Stat.Spec
...
+  let nativeSpecs = tests session
+  asTastySpecs <- Tasty.testSpecs $ legacySpecs session
+  let allSpecs = nativeSpecs <> asTastySpecs
+  pure . Tasty.localOption Tasty.Success $ testGroup "semantic" allSpecs
+
+-- If you're writing new test modules, please don't add to this
+-- stanza: it is only there to prevent massive rewrites, and is
+-- converted into a Tasty TestTree in 'main'. (Quoth the tasty-hspec
+-- documentation: "hspec and tasty serve similar purposes; consider
+-- using one or the other.") Instead,
```
Comment (robrix) [inline test/Spec.hs]: 1. Thank you for these docs.
2. Final line seems to be cut off.

File: test/fixtures/json/corpus/hash.diffA-B.txt [hunk 5]
```diff
@@ -1,7 +1,7 @@
 (Hash
-  {-(KeyValue
-    {-(TextElement)-}
-    {-(Float)-})-}
+{-(KeyValue
```
[the remark this answers is not in the document]
Comment (robrix) [reply test/fixtures/json/corpus/hash.diffA-B.txt]: A feature, IMO.

File: test/Integration/Spec.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (robrix) [reply test/Integration/Spec.hs]: (I kid, I kid. I’m cool with this 👍)

File: semantic.cabal (hunk 2 above)
Comment (patrickt) [reply semantic.cabal]: Not technically needed, but ansi-terminal just hit 0.9.0, and the old `hspec-expectations-pretty-diff` was complaining about it.

File: semantic.cabal (hunk 3 above)
Comment (patrickt) [reply semantic.cabal]: Yup. We should really just turn on -Wall everywhere, but I’ll file a separate issue about that.

File: test/Spec.hs (hunk 4 above)
Comment (patrickt) [reply test/Spec.hs]: Oops

File: semantic.cabal (hunk 3 above)
Comment (robrix) [reply semantic.cabal]: I dig it; the thing `fused-effects` does is working pretty nicely.

File: semantic.cabal (hunk 2 above)
Comment (robrix) [reply semantic.cabal]: Gotcha.

**Pull request merged**